internal/bminventory/inventory.go

```diff
@@ -2627,7 +2627,7 @@ func (b *bareMetalInventory) uploadLogs(ctx context.Context, params installer.Up
 
 	_, err := b.getCluster(ctx, params.ClusterID.String())
 	if err != nil {
-		common.GenerateErrorResponder(err)
+		return err
```

Comment:
Same issue on line 2637?
`return common.NewApiError(http.StatusInternalServerError, err)`

Reply:
No, that's fine. `ApiResponseError` fulfills the `error` interface. This line had two problems. Missing `return` and `middleware.Responder` is not an `error`.